packages/sails-ng-common/src/validation/validators.ts

```diff
@@ -305,4 +298,39 @@ export const formValidatorsSharedDefinitions: FormValidatorDefinition[] = [
       };
     },
   },
+  {
+    name: "jsonata-expression",
+    message: "@validator-error-jsonata-expression",
+    create: (config) => {
+      const optionNameKey = "name";
+      const optionNameValue = formValidatorGetDefinitionString(config, optionNameKey, "jsonata-expression");
+      const optionMessageKey = "message";
+      const optionMessageValue = formValidatorGetDefinitionString(config, optionMessageKey, "@validator-error-jsonata-expression");
+      const optionDescriptionKey = "description";
+      const optionDescriptionValue = formValidatorGetDefinitionString(config, optionDescriptionKey);
+      const optionExpressionKey = "expression";
+      const expression = formValidatorGetDefinitionItem(config, optionExpressionKey);
+      const optionEvaluatorKey = "evaluator";
+      const evaluator = formValidatorGetDefinitionItem(config, optionEvaluatorKey) as (control: FormValidatorControl) => boolean;
+      return (control) => {
+          if (control.value == null || formValidatorLengthOrSize(control.value) === 0) {
+              return null; // don't validate empty values to allow optional controls
+          }
+          const value = control.value?.toString();
+          const success = evaluator(control)
```

Comment:
Probably should try catch this, print the error then return false? 
If the scripts broken then there's no good scenario but it's probably better to report to the user the vaildation error than to break the form.